### DISCUSSION TITLE
Feedback on download export zip file

### DISCUSSION THREAD

**Discussion by skttl** [umbraco/Umbraco.Deploy.Issues#322] [Ideas]
Trying to export a complete workspace, for importing in to a new site (v8 to vLatest upgrade).

I could use some visual feedback about what is happening in the dialog. There is just a button that is spinning. The export is huge, and I can see in the network tab that it is downloading though.

![image](https://github.com/umbraco/Umbraco.Deploy.Issues/assets/3726467/3620bb84-594e-4c8d-93af-bc8ffc9d0f3d)

Also, if something breaks (connection outage etc), I don't seem to have any way of getting the zip file again, other than exporting a new one. Is that how it's meant to be? There could be a list somewhere of exports, where I can also delete the ones not to be used anymore.

**Comment by skttl** [comment]
And the zip file I ended up getting was only 1KB? I saw it downloading more than a gigabyte in my dev tools.

**Comment by ronaldbarendse** [comment]
Thanks for reporting this Søren!

Deploy uses `umbRequestHelper.downloadFile()` (in v8-v13) provided by the CMS to initiate the file download and get updates on progress/completion, but that uses a temporary browser [blob buffer](https://developer.mozilla.org/en-US/docs/Web/API/Blob) that seems to have [browser/OS specific limits](https://stackoverflow.com/a/43816041/96175). I've successfully downloaded (and even uploaded) 1.5GB ZIP archives, which seems to be within the 2GB Chrome on Windows x64 limit of my PC...

We do recommend a different method for transferring large amounts on media though, like a direct Azure Blob Storage copy or using FTP to download/upload the media files (see [docs](https://docs.umbraco.com/umbraco-deploy/v/13.deploy.latest-lts/deployment-workflow/import-export#exporting-content-and-schema)).

As a workaround, you can also try opening the `.../DownloadExport?id=` request in a new tab (from the Dev Tools), as that should result in a regular download (without the use of the blob buffer). We'll investigate whether we can avoid using the buffer and invoke a direct download in the UI while keeping the progress updates.

**Comment by hfloyd** [comment]
I agree. The Export "download" just doesn't seem to work.

These are the Console errors I get:
![image](https://github.com/user-attachments/assets/077dfd8b-02f4-469b-8633-5b81690c4f7b)

Umbraco Cloud - Cms 13.5.2 - Forms 13.2.5 - Deploy 13.2.2

If the file was stored in Kudu or something, it might be easier to retrieve?

@ronaldbarendse Thanks for the tip about opening the link in a new tab - that actually results in a download box:
![image](https://github.com/user-attachments/assets/ba556853-941c-41a5-aa5a-87a3b0d63ca3)


Though, even if this huge file eventually gets downloaded it seems I won't be able to use it to update the empty dev/staging sites...
![image](https://github.com/user-attachments/assets/bb22a835-e8a0-4850-8ab9-eff511b12cb0)

[Insert picture of Heather banging her head against keyboard here]

**Comment by hfloyd** [comment]
Ah, yep... At first I was hopeful:
![image](https://github.com/user-attachments/assets/3861734d-a112-4a22-87a9-e23f1c98e336)

But, ultimately, disappointed:
![image](https://github.com/user-attachments/assets/b981a5c7-ada8-41a0-9ba3-0110f91d01d8)

**Comment by hfloyd** [comment]
My next attempt (based on Ronald's previous recommendation) was more sucessful:

1. Use Azure Storage Explorer to connect to Live site media files. 
2. Download "media" folder to local machine.
3. Use Azure Storage Explorer to connect to empty Staging & Development sites, and upload the downloaded media folder.
4. In Live site back-office, create a much smaller Export file using only the first option: ![image](https://github.com/user-attachments/assets/20087fa0-7e2a-427f-9a29-a578422eaa44)
5. Import the file into the Staging and Development back-offices.

PS. It would be nice if the stored Export files were accessible for Download here as well as deletion - since sometimes the page gets frozen, or accidentally refreshed, and the download button is no longer present.
![image](https://github.com/user-attachments/assets/52568d4e-43f8-4bbd-8c5d-a0b6a9ee7486)

**Comment by umbracocommunity** [comment]
This issue has been mentioned on **Umbraco community forum**. There might be relevant details there:

https://forum.umbraco.com/t/umbraco-version-8-18-15-workspace-export-doesnt-work/784/8
